tests/components/gree/test_climate.py

```diff
@@ -289,14 +291,14 @@ async def test_send_power_on(hass, discovery, device, mock_now):
 
     assert await hass.services.async_call(
         DOMAIN,
-        SERVICE_SET_HVAC_MODE,
-        {ATTR_ENTITY_ID: ENTITY_ID, ATTR_HVAC_MODE: HVAC_MODE_AUTO},
+        SERVICE_TURN_ON,
+        {ATTR_ENTITY_ID: ENTITY_ID},
         blocking=True,
     )
 
     state = hass.states.get(ENTITY_ID)
     assert state is not None
-    assert state.state == HVAC_MODE_AUTO
+    assert state.state != HVAC_MODE_OFF
```

Comment:
Maybe this test should stay in place, this service was not removed, now was it ?

Reply:
The tes isn’t valid in it’s previous state. Gree device use two separate variables for power state and mode. When sending a command to turn on the device the only reliable assertion we can make is that the mode will not be off.

Reply:
Sorry... deleted last response as I am trying to remember work I did here something like 2 months ago now.

This test is a power on test, the previous assertion wasn’t valid and introducing these service calls made that apparent. Since Gree uses two separate variables for mode  and power state, when sending power on the only reliable assertion is that it isn’t off.

In my previous implementation I didn’t really handle his well, so the test was carried out by setting a mode which implied on. And in fact all it was really doing in the end was testing that a set mode was applied correctly, the test for modes is carried out further down, so this case still remains covered.